packages/language-server/src/plugins/typescript/TypeScriptPlugin.ts

```diff
@@ -329,6 +330,31 @@ export class TypeScriptPlugin
         const { lang, tsDoc } = await this.getLSAndTSDoc(document);
         const mainFragment = tsDoc.getFragment();
 
+        const cssClassHelper = new CSSClassDefinitionLocator(tsDoc, position, document);
+        const cssDefinitionRange = cssClassHelper.getCSSClassDefinition();
+        if (cssDefinitionRange) {
+            const results: DefinitionLink[] = [];
+            cssDefinitionRange.start.character++; //Report start of name instead of start at . for easy rename (F2) possibilities
+
+            const originRange = Range.create(
+                Position.create(position.line, position.character),
+                Position.create(position.line, position.character)
+            );
+
+            results.push(
+                LocationLink.create(
+                    pathToUrl(document.getFilePath() as string),
+                    cssDefinitionRange,
+                    cssDefinitionRange,
+                    originRange
+                )
+            );
+
+            if (results) {
+                return results;
```

Comment:
The results should probably be merged with what else comes up. In the case of `class:foo` both `.foo { ... }` in `<style>` and `let foo = ...` in `script` are definitons, and we would lose the latter when returning early.

Reply:
I'm not sure I follow. In all the tests and variations I've used it seems to work. When you hit F12 to go to definition there can only be 1 result (if a definition can be found).